### PR TITLE
Add bounds checking on incoming netplay packet

### PR DESCRIPTION
**Pull request by altf4**
- Also removed panicalert on unknown message type

### CODE REVIEW

File: Source/Core/Core/Slippi/SlippiNetplay.cpp [hunk 1]
```diff
@@ -172,6 +172,13 @@ unsigned int SlippiNetplayClient::OnData(sf::Packet &packet)
 
 			int32_t headFrame = remotePadQueue.empty() ? 0 : remotePadQueue.front()->frame;
 			int inputsToCopy = frame - headFrame;
+
+			// Check that the packet actually contains the data it claims to
+			if((5 + inputsToCopy * SLIPPI_PAD_DATA_SIZE) > (int)packet.getDataSize())
+			{
```
Comment (JLaferri) [inline Source/Core/Core/Slippi/SlippiNetplay.cpp:186]: Add error logger here. This case should really never happen so error logger probably makes sense.

File: Source/Core/Core/Slippi/SlippiNetplay.cpp [hunk 2]
```diff
@@ -172,6 +172,13 @@ unsigned int SlippiNetplayClient::OnData(sf::Packet &packet)
 
 			int32_t headFrame = remotePadQueue.empty() ? 0 : remotePadQueue.front()->frame;
 			int inputsToCopy = frame - headFrame;
+
+			// Check that the packet actually contains the data it claims to
+			if((5 + inputsToCopy * SLIPPI_PAD_DATA_SIZE) > (int)packet.getDataSize())
```
Comment (JLaferri) [inline Source/Core/Core/Slippi/SlippiNetplay.cpp:185]: Would it perhaps be better to put this check in the loop instead? That way if there is actually some good data it gets extracted. Probably not a major concern seeing as how this case is effectively not handled anyway but just a thought.

**Review by JLaferri — CHANGES_REQUESTED** [review]

File: Source/Core/Core/Slippi/SlippiNetplay.cpp (hunk 2 above)
Comment (altf4) [reply Source/Core/Core/Slippi/SlippiNetplay.cpp:185]: It's probably possible but... I'd be afraid of getting the ack wrong. Like, if a packet comes in that should have 5 pads, but only has space for 2.5, you have to ack only the pads that you've actually read. But the individual pads aren't labeled with a frame. (the frame number is inferred from how it comes in)

Basically, if a packet somehow got chopped in half, then we don't have a good way to know which half we're reading. Seems safer to just reject the packet, imo.

File: Source/Core/Core/Slippi/SlippiNetplay.cpp (hunk 1 above)
Comment (altf4) [reply Source/Core/Core/Slippi/SlippiNetplay.cpp:186]: Added

**Pull request merged**